onnxruntime/contrib_ops/cpu/murmur_hash3.cc

```diff
@@ -171,34 +171,42 @@ Status MurmurHash3::Compute(OpKernelContext* ctx) const {
   Tensor* output_tensor = ctx->Output(0, input_shape);
 
   const MLDataType keys_type = keys->DataType();
-  const int input_element_bytes = static_cast<int>(keys->DataType()->Size());
-  const int output_element_bytes = static_cast<int>(output_tensor->DataType()->Size());
+  const bool is_string = keys_type == DataTypeImpl::GetType<std::string>();
+  if (!is_string && (DataTypeImpl::GetType<int32_t>() != keys_type) && 
+      (DataTypeImpl::GetType<uint32_t>() != keys_type)) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, NOT_IMPLEMENTED, "Type not supported.");
+  }
```

Comment:
Doesn't the constraints checker in Graph (or maybe onnx) do this based on the schema? 

Same question with validating the output type - I thought that was handled by generic code based on the constraints in the schema.

Reply:
I agree with you. This was in the original code.  Not sure about the output type though.

---
In reply to: [338823167](https://github.com/microsoft/onnxruntime/pull/2257#discussion_r338823167) [](ancestors = 338823167)